test/distributed/_tensor/test_math_ops.py

```diff
@@ -200,7 +209,100 @@ def _replicate_fn(name, module, device_mesh):
+            self.assertEqual(y_local, y_dist.full_tensor())
+
+    @with_comms
+    def test_layer_norm_bwd(self):
+        device_mesh = self.build_device_mesh()
+
+        # NLP example from pytorch docs
+        # https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html
+        batch, sentence_length, embedding_dim = 20, 5, 10
+        norm_shape_idx_list = list(range(3))
+        shard_dims = [-1, 0, 1, 2]
+        elementwise_affine_list = [False, True]
+        test_config_list = list(
+            itertools.product(shard_dims, norm_shape_idx_list, elementwise_affine_list)
+        )
+
+        # normalized shape is a torch.Size object
+        for shard_dim, norm_idx, elementwise_affine in test_config_list:
+            x = torch.rand(
+                batch,
+                sentence_length,
+                embedding_dim,
+                device=self.device_type,
+                requires_grad=True,
+            )
+            normalized_shape = x.shape[norm_idx:]
+            layer_norm = torch.nn.LayerNorm(
+                normalized_shape,
+                elementwise_affine=elementwise_affine,
+                device=self.device_type,
+            )
+            layer_norm_local = copy.deepcopy(layer_norm).to(self.device_type)
+
+            def _replicate_fn(name, module, device_mesh):
+                for name, param in module.named_parameters():
+                    if name in ["weight", "bias"]:
+                        param_dist = torch.nn.Parameter(
+                            distribute_tensor(param, device_mesh, [Replicate()])
+                        )
+                        module.register_parameter(name, param_dist)
+
+            layer_norm_dist = distribute_module(layer_norm, device_mesh, _replicate_fn)
+
+            if elementwise_affine:
+                self.assertEqual(
+                    layer_norm_local.weight, layer_norm_dist.weight.full_tensor()
+                )
+                self.assertEqual(
+                    layer_norm_local.bias, layer_norm_dist.bias.full_tensor()
+                )
+
+            x_local = x.detach().clone().requires_grad_(True)
+            x_dist = distribute_tensor(x, device_mesh, [Shard(shard_dim)])
+            self.assertEqual(x_local, x_dist.full_tensor())
+
+            y_local = layer_norm_local(x_local)
+            # make sure that backward layer norm does not introduce extra collectives
+            comm_mode = CommDebugMode()
+            with comm_mode:
+                y_dist = layer_norm_dist(x_dist)
+
+            self.assertEqual(
+                comm_mode.get_total_counts(),
+                0,
+                f"comm count={comm_mode.get_total_counts()}, "
+                f"shard_dim={shard_dim}, norm_shape={normalized_shape}, elem_affine={elementwise_affine}",
+            )
+
+            self.assertEqual(y_local, y_dist.full_tensor())
+
+            # backward step
+            comm_mode = CommDebugMode()
```

Comment:
I think you can reuse the `comm_mode` initialize earlier instead of creating a new one, the collective counts will be cleared for a new context